### ISSUE TITLE
Allow extended properties to be logged into influxdb

### ISSUE THREAD

**Issue by simwillo** [pada57/serilog-sinks-influxdb#15]
I would like to add values to the properties collection of a LogEvent and then have these items written to influxdb. I propose that the influxDBSinkOptions class should be extended to allow a list of fields to be defined

```
public class InfluxDBSinkOptions
{
public string ApplicationName { get; set; }

    public string InstanceName { get; set; }

    public InfluxDBConnectionInfo ConnectionInfo { get; set; } = new InfluxDBConnectionInfo();

    public PeriodicBatchingSinkOptions BatchOptions { get; set; } = new PeriodicBatchingSinkOptions();

    public IFormatProvider FormatProvider { get; set; } = null;

    **public string[] LoggedProperties { get; set; }**

}
This should then be emitted by the InfluxDBSink class

   public async Task EmitBatchAsync(IEnumerable<LogEvent> batch)
    {
        if (batch == null) throw new ArgumentNullException(nameof(batch));

        var logEvents = batch as List<LogEvent> ?? batch.ToList();
        var points = new List<Point>(logEvents.Count);

        foreach (var logEvent in logEvents)
        {
            var p = new Point
            {
                Name = PointName,
                Fields = new Dictionary<string, object>(),
                Timestamp = logEvent.Timestamp.UtcDateTime
            };

            // Add tags
            if (logEvent.Exception != null) p.Tags.Add("exceptionType", logEvent.Exception.GetType().Name);

            var severity = logEvent.Level.ToSeverity();

            p.Tags.Add(Tags.Level, logEvent.Level.ToString());
            p.Tags.Add(Tags.AppName, _applicationName);
            p.Tags.Add(Tags.Facility, _instanceName);
            p.Tags.Add(Tags.Hostname, Environment.MachineName);
            p.Tags.Add(Tags.Severity, severity.ToString());

            **foreach (var loggedProperty in _loggedProperties) {
                if (logEvent.Properties.ContainsKey(loggedProperty)) {
                    p.Tags.Add(loggedProperty, logEvent.Properties[loggedProperty]);
                }
            }**
            
            // Add Fields - rendered message
            p.Fields[Fields.Message] = StripSpecialCharacter(logEvent.RenderMessage(_formatProvider));
            p.Fields[Fields.Facility] = 16;
            p.Fields[Fields.ProcId] = Process.GetCurrentProcess().Id;
            p.Fields[Fields.Severity] = severity.ToString();
            p.Fields[Fields.Timestamp] = logEvent.Timestamp.ToUnixTimeMilliseconds() * 1000000;
            p.Fields[Fields.Version] = 1;

            points.Add(p);
        }

**Comment by pada57** [comment]
@simwillo  yes could be something we could add. For fields i followed the syslog format. For tags it might be high cardinality for the values of your new logged event properties which might be an issue as Tags.

do you want to add them as field or tags ? we might renamed this to Extended tags and Extended fields which might be more understandable

**Comment by simwillo** [comment]
@pada57 working on the basis that tags differ from fields in that tags are indexed, in our use case maybe tags would be better, but I think your suggestion of having both ExtendedTags and ExtendedFileds makes sense.

**Comment by simwillo** [comment]
@pada57 will prepare a PR

**Comment by simwillo** [comment]
@pada57 Apologies for the delay in submitting the PR, but would you be willing to except PRs for version 1 of the serilog-sinks

**Comment by pada57** [comment]
@simwillo ok you want same feature with extended properties and tags which was added in V2 to v1 ?

**Comment by simwillo** [comment]
Yes that is right

**Comment by tom-englert** [comment]
@pada57 looking at the existing code this seems to be already resolved:

https://github.com/pada57/serilog-sinks-influxdb/blob/f894b193932427e1a5aded816d805fd42c46a5d5/Serilog.Sinks.InfluxDB/Sinks/InfluxDB/InfluxDBSink.cs#L114-L117

also #17 seems to be superseded by this.

**Comment by pada57** [comment]
fixed by #17
